src/Storefront/Resources/views/storefront/base.html.twig

```diff
@@ -36,44 +36,43 @@
             </noscript>
         {% endblock %}
 
-        {% block base_header %}
-            {% sw_include '@Storefront/storefront/utilities/staging-info.html.twig' %}
-
-            <header class="header-main">
-                {% block base_header_inner %}
-                    <div class="container">
-                        {% sw_include '@Storefront/storefront/layout/header/header.html.twig' %}
-                    </div>
-                {% endblock %}
-            </header>
-        {% endblock %}
-
-        {% block base_navigation %}
-            {% if feature('v6.7.0.0') %}
-                <div class="nav-main d-none d-lg-block">
-                    {% block base_navbar_inner %}
-                        {% sw_include '@Storefront/storefront/layout/navbar/navbar.html.twig' %}
+        {% if feature('cache_rework') %}
+            {{ render_esi(url('frontend.header', { 'navigationId': shopware.navigation.id })) }}
```

Comment:
I am not sure, if this could contradict the whole purpose of ESI a bit :grimacing: 

If I understand that correctly, we would now have a cache for each navigation ID :thinking: 

for now, it was the easiest solution, to fix the offcanvas menu. but there might be another solution, if this is a problem

Reply:
We need a JS plugin-based solution, so the JS plugin selects the active category.
And yes, currently, this looks like generating a cache for every navigation ID.

Reply:
yep the original idea was resolve the active navigation on the client

Not sure how far that implementation was in the POC, @tobiasberge didn't you pair up to try some things for that?

Reply:
I think, I figured out how the JS plugin needs to be adjusted. please have a look again :blush: